.ci/Jenkinsfile

```diff
@@ -0,0 +1,79 @@
+#!/usr/bin/env groovy
+
+// Verified with Jenkins v2.164.3
+
+// TODO:
+// 1. Calculate taskset/affinity for the scripts based on total number of jenkins executors
+
+pipeline {
+    agent {label "ml-test-node-gpu"}
+
+    options {
+        buildDiscarder(logRotator(numToKeepStr:'10'))
+        timeout(time: 15, unit: 'MINUTES')
+        disableConcurrentBuilds()
+    }
+
+    environment {
+        ARTEFACT_DIR = "${WORKSPACE}/artefacts"
+        NFS_WORKSPACE = "${NFS_WORKSPACE_ROOT}/ml-nccl-rdma-sharp-plugins-pr/${BUILD_NUMBER}"
+        NCCL_SRC_DIR = "${WORKSPACE}/nccl"
+        NCCL_DIR = "${NFS_WORKSPACE}/nccl"
+        NCCL_TESTS_DIR = "${NFS_WORKSPACE}/nccl-tests"
+    }
+
+    stages {
+        stage('Preparations') {
+            steps {
+                echo 'Preparations...'
+                sh 'mkdir -p ${ARTEFACT_DIR}'
+                sh 'mkdir -p ${NFS_WORKSPACE}'
+            }
+        }
+        stage('Checkout NCCL') {
+            steps {
+                sh 'mkdir -p ${NCCL_SRC_DIR}'
+                dir("${NCCL_SRC_DIR}") {
+                    git branch: 'master',
+                    url: 'git@gitlab.com:Mellanox/nccl-sharp.git'
```

Comment:
would externalize branch name and url into envitonment section
at some point it will move to upstream and will need to support various branches from upstream.
in env section you can change values w/o touching code.

Reply:
Fixed.